cloudinit/cmd/main.py

```diff
@@ -1273,15 +1277,15 @@ def sub_main(args):
     report_on = True
     if name == "init":
         if args.local:
-            rname, rdesc = ("init-local", "searching for local datasources")
+            rname, rdesc = ("local", "searching for local datasources")
```

Comment:
I'm fine with keeping the originals for now. The internal references I don't care about as much, but it would benefit users if long-term we got to the point where user-facing references to stages used the same vocabulary as our docs.

I'm sure users find this sequence of logs a bit confusing:
```
2024-05-16 16:59:04,331 - util.py[DEBUG]: cloud-init mode 'init' took 0.362 seconds (0.36)
2024-05-16 16:59:04,331 - handlers.py[DEBUG]: finish: init-local: SUCCESS: searching for local datasources
2024-05-16 16:59:04,778 - util.py[DEBUG]: Cloud-init v. 24.1.3-0ubuntu3 running 'init' at Thu, 16 May 2024 16:59:04 +0000. Up 13.32 seconds.
```

This is confusing:

1. Why did 'init' mode end only to start again?
2. Is init-local different than 'init'?

Reply:
@TheRealFalcon I dropped this commit for now (out of scope of this PR), but do note that `name` was replaced by `rname` a few lines lower in this file.